### ISSUE TITLE
0.15.2 to_datetime broken 

### ISSUE THREAD

**Issue by bjmacdonald** [pydata/pandas#9405]
It looks like the support for the new 'exact' keyword has borked to_datetime() for format strings other than '%Y%m%d' as it is passed to array_strptime which doesn't seem to support it:

    ts = 201502031753
    
    pd.to_datetime(ts, format='%Y%m%d%H%M')
    ---------------------------------------------------------------------------
    TypeError                                 Traceback (most recent call last)
    <ipython-input-2-685333561e5e> in <module>()
    ----> 1 pd.to_datetime(ts, format='%Y%m%d%H%M')
    
    C:\app\python\lib\site-packages\pandas\tseries\tools.py in to_datetime(arg, errors, dayfirst, utc, box, format, exact, coerce, unit, infer_datetime_format)
        316         return _convert_listlike(arg, box, format)
        317 
    --> 318     return _convert_listlike(np.array([ arg ]), box, format)[0]
        319 
        320 class DateParseError(ValueError):
    
    C:\app\python\lib\site-packages\pandas\tseries\tools.py in _convert_listlike(arg, box, format)
        278                     try:
        279                         result = tslib.array_strptime(
    --> 280                             arg, format, exact=exact, coerce=coerce
        281                         )
        282                     except (tslib.OutOfBoundsDatetime):
    
    C:\app\python\lib\site-packages\pandas\tslib.pyd in pandas.tslib.array_strptime (pandas\tslib.c:35070)()
    
    TypeError: array_strptime() got an unexpected keyword argument 'exact'

**Comment by jorisvandenbossche** [comment]
Are you sure you are running 0.15.2? But in any case it is a strange error message (that you as a user never should see). 

For me it works as expected:

```
In [8]: ts = 201502031753

In [9]: pd.to_datetime(ts, format='%Y%m%d%H%M')
Out[9]: Timestamp('2015-02-03 17:53:00')

In [10]: pd.__version__
Out[10]: '0.15.2'
```

**Comment by jreback** [comment]
it looks like you don't have the correct compiled code - pls clarify how you installed

**Comment by bjmacdonald** [comment]
Installed via Continuum Anaconda python 3.4 on Windows 64bit.  Not the first time I've had problems with their python3 support.  Version is definitely 0.15.2.  I will open an issue with them.

**Comment by TomAugspurger** [comment]
FWIW, I have the same setup and it parses correctly for me. I did have an issue when I upgraded to .15.1 where I had to clear pandas from the cache in the Pkgs directory under the Conda home, and then reinstall.

**Comment by bjmacdonald** [comment]
I have now established it was a problem with  'conda update' which silently fails to copy the pyd files if they are locked by a running process!  Thanks for the feedback.


-------- Original message --------
From: Tom Augspurger
Date:04/02/2015 13:07 (GMT+00:00)
To: pydata/pandas
Cc: Bruce MacDonald
Subject: Re: [pandas] 0.15.2 to_datetime broken (#9405)


FWIW, I have the same setup and it parses correctly for me. I did have an issue when I upgraded to .15.1 where I had to clear pandas from the cache in the Pkgs directory under the Conda home, and then reinstall.

-
